pkg/app/two_pass_renderer.go

```diff
@@ -116,15 +116,13 @@ func (r *desiredStateLoader) twoPassRenderTemplateToYaml(inherited, overrode *en
 		r.logger.Debugf("first-pass rendering result of \"%s\": %v", filename, *finalEnv)
 	}
 
-	vals := map[string]interface{}{}
-	if prestate != nil {
-		prestate.Env = *finalEnv
-		vals, err = prestate.Values()
-		if err != nil {
-			return nil, err
-		}
+	vals, err := finalEnv.GetMergedValues()
+	if err != nil {
+		return nil, err
 	}
+
 	if prestate != nil {
+		prestate.Env = *finalEnv
```

Comment:
Just asking - do you by any chance figure out why we mutate prestate here? (I lost my memory and I understand you've kept this in favor of the L121 before this change.

Reply:
I think we can safely skip the mutation actually - we don't really continue to use the prestate past this point;
I kept it b/c I wasn't sure if the mutation was intentional or not - didn't want to break anything (: